contracts/utils/cryptography/draft-ERC7739Signer.sol

```diff
@@ -55,21 +55,28 @@ abstract contract ERC7739Signer is EIP712, IERC1271 {
 
     /**
      * @dev Nested personal signature verification.
+     *
+     * NOTE: Instead of overriding this function, try with {_validateNestedEIP712Signature}. It encapsulates
+     * nested EIP-712 hashes.
      */
-    function _isValidNestedPersonalSignSignature(
-        bytes32 hash,
-        bytes calldata signature
-    ) internal view virtual returns (bool) {
-        return _validateSignature(_domainSeparatorV4().toTypedDataHash(hash.personalSignStructHash()), signature);
+    function _isValidNestedPersonalSignSignature(bytes32 hash, bytes calldata signature) internal view returns (bool) {
+        return
+            _validateNestedEIP712Signature(
+                _domainSeparatorV4().toTypedDataHash(hash.personalSignStructHash()),
+                signature
+            );
     }
 
     /**
      * @dev Nested EIP-712 typed data verification.
+     *
+     * NOTE: Instead of overriding this function, try with {_validateNestedEIP712Signature}. It encapsulates
+     * nested EIP-712 hashes.
      */
     function _isValidNestedTypedDataSignature(
         bytes32 hash,
         bytes calldata encodedSignature
-    ) internal view virtual returns (bool) {
+    ) internal view returns (bool) {
```

Comment:
Our policity is to make every internal function virtual, unless we have a very strong reason against it. is there any serious issues/danger in having this function virtual ?

Reply:
There's no issue/danger, and I wanted to experiment with the idea of isolating the implementation, I think those two functions shouldn't customized and instead developers should either override `_validateSignature` or `_isValidSignature`.

What do you think of making both `_isValidNestedTypedDataSignature` and `_isValidNestedPersonalSignSignature` private instead?

I'm reverting for now

Reply:
making them private is prossibly the right move.